### PR TITLE
Alphabetically sorting CLI completions. JIRA #2356

### PR DESCRIPTION
**Pull request by Shilpaks**

### CODE REVIEW

File: common-cli/src/main/java/co/cask/common/cli/completers/StringsCompleter.java [hunk 1]
```diff
@@ -36,13 +39,14 @@
   @Override
   protected Collection<String> getCandidates(String buffer) {
     ImmutableList.Builder<String> candidates = ImmutableList.builder();
-
+    ArrayList<String> mutableCandidates = new ArrayList<String>();
```
Comment (chtyim) [inline common-cli/src/main/java/co/cask/common/cli/completers/StringsCompleter.java]: Use `List` as type.

File: common-cli/src/main/java/co/cask/common/cli/completers/StringsCompleter.java [hunk 2]
```diff
@@ -36,13 +39,14 @@
...
     for (String candidate : getStrings()) {
       if (candidate.startsWith(buffer)) {
-        candidates.add(candidate);
+        mutableCandidates.add(candidate);
       }
     }
-
+    Collections.sort(mutableCandidates, Ordering.usingToString());
+    candidates.addAll(mutableCandidates);
```
Comment (chtyim) [inline common-cli/src/main/java/co/cask/common/cli/completers/StringsCompleter.java]: Copying to an immutable list is unnecessary. Just have

`return Collections.unmodifiableList(mutableCandidates);`

**Comment by chtyim** [comment]
Just two minor comments. Rest LGTM.

One side note about commit message. Usually we prefix the message with the issue number. E.g.

`(CDAP-2356) Sort CLI auto-completion alphabetically.`

File: common-cli/src/main/java/co/cask/common/cli/completers/StringsCompleter.java (hunk 1 above)
Comment (bdmogal) [inline common-cli/src/main/java/co/cask/common/cli/completers/StringsCompleter.java]: for instantiation, use `Lists.newArrayList()` as opposed to `new ArrayList<String>()`

File: common-cli/src/test/java/co/cask/common/cli/completers/StringsCompleterTest.java [hunk 3]
```diff
@@ -19,9 +19,12 @@
 import com.google.common.base.Supplier;
 import com.google.common.base.Suppliers;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
 import org.junit.Test;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
```
Comment (gsps1) [inline common-cli/src/test/java/co/cask/common/cli/completers/StringsCompleterTest.java]: unused imports ?

File: common-cli/src/test/java/co/cask/common/cli/completers/StringsCompleterTest.java [hunk 4]
```diff
@@ -37,9 +40,12 @@ public void testComplete() {
       }
     };
 
-    testCompleter(completer, "a", 0, ImmutableList.<CharSequence>of("asdf", "asdd"));
+    testCompleter(completer, "a", 0, ImmutableList.<CharSequence>of("asdd", "asdf"));
     testCompleter(completer, "b", 0, ImmutableList.<CharSequence>of("bdf"));
     testCompleter(completer, "c", 0, ImmutableList.<CharSequence>of());
+    testOrder(completer, "a", 0, ImmutableList.<CharSequence>of("asdd", "asdf"));
+
+
```
Comment (gsps1) [inline common-cli/src/test/java/co/cask/common/cli/completers/StringsCompleterTest.java]: remove unwanted space

**Pull request merged**